src/android/CameraPreviewFragment.java

```diff
@@ -1,109 +1,208 @@
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        lifecycleRegistry = new LifecycleRegistry(this::getLifecycle);
+        lifecycleRegistry.setCurrentState(Lifecycle.State.CREATED);
+
         RelativeLayout containerView = new RelativeLayout(getActivity());
         RelativeLayout.LayoutParams containerLayoutParams = new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT);
         containerLayoutParams.addRule(RelativeLayout.ALIGN_PARENT_TOP);
         containerLayoutParams.addRule(RelativeLayout.ALIGN_PARENT_START);
         containerView.setLayoutParams(containerLayoutParams);
-        camera = new CameraView(getActivity());
-        camera.setAudio(Audio.OFF);
-        camera.addCameraListener(new CameraListener() {
-            @Override
-            public void onCameraOpened(CameraOptions options) {
-                if (startCameraCallback != null)
-                    startCameraCallback.onCameraStarted();
-            }
-            @Override
-            public void onPictureTaken(PictureResult result) {
-                UUID uuid = UUID.randomUUID();
-                File file = new File(new ContextWrapper(getActivity().getBaseContext()).getFilesDir(), uuid.toString() + ".jpg");
-                result.toFile(file, (File mfile)->{
-                    if (mfile == null){
-                        capturePictureCallback.onCompleted(new Exception("unable to save image"), null);
-                    }
-                    capturePictureCallback.onCompleted(null, mfile.getName());
-                });
 
-            }
-        });
-        camera.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT));
-        containerView.addView(camera);
+        viewFinder = new PreviewView(getActivity());
+        viewFinder.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT));
+        containerView.addView(viewFinder);
+        startCamera();
+
         return containerView;
     }
 
     @Override
-    public void onResume() {
-        super.onResume();
-        camera.open();
+    public void onStart() {
+        super.onStart();
+
+        if (lifecycleRegistry.getCurrentState() == Lifecycle.State.CREATED) {
+            lifecycleRegistry.setCurrentState(Lifecycle.State.STARTED);
+        }
+    }
+
+    public void startCamera() {
+        ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(getActivity());
+        ProcessCameraProvider cameraProvider = null;
+
+        try {
+            cameraProvider = cameraProviderFuture.get();
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            Log.e(TAG, "startCamera: " + e.getMessage());
```

Comment:
@dinitri-ragoo here we should return using a failure callback to the javascript context ?

Reply:
I'm not sure whether it is possible or not to return a failure callback from this method. As it has no direct relation to the javascript context

Reply:
@dinitri-ragoo @azharbeebeejaun we should discuss this tomorrow.